tool.ts

```diff
@@ -84,33 +103,45 @@ export function evaluateVersions(versions: string[], versionSpec: string): strin
 /**
  * finds the path to a tool in the local installed tool cache
  * 
- * @param toolName  name of the tool
- * @param version   version to get the path of
- * @param arch      optional arch.  defaults to arch of computer
+ * @param toolName      name of the tool
+ * @param versionSpec   version of the tool
+ * @param arch          optional arch.  defaults to arch of computer
  */
-export function findLocalTool(toolName: string, version: string, arch?: string) {
+export function findLocalTool(toolName: string, versionSpec: string, arch?: string): string {
     if (!toolName) {
         throw new Error('toolName parameter is required');
     }
 
-    if (!version) {
-        throw new Error('version parameter is required');
+    if (!versionSpec) {
+        throw new Error('versionSpec parameter is required');
     }
 
-    let cacheRoot = _getCacheRoot();
-
     arch = arch || os.arch();
 
-    let installedPath: string;
-    let cachePath = path.join(cacheRoot, toolName, semver.clean(version), arch);
-    tl.debug('cachePath: ' + cachePath);
+    // attempt to resolve an explicit version
+    if (!isExplicitVersion(versionSpec)) {
+        let localVersions: string[] = findLocalToolVersions(toolName, arch);
+        let match = evaluateVersions(localVersions, versionSpec);
+        versionSpec = match;
+    }
 
-    if (tl.exist(cachePath) && tl.exist(`${cachePath}.complete`)) {
-        installedPath = cachePath;
+    // check for the explicit version in the cache
+    let toolPath: string;
+    if (versionSpec) {
+        versionSpec = semver.clean(versionSpec);
+        let cacheRoot = _getCacheRoot();
+        let cachePath = path.join(cacheRoot, toolName, versionSpec, arch);
+        tl.debug('checking cache: ' + cachePath);
+        if (tl.exist(cachePath) && tl.exist(`${cachePath}.complete`)) {
+            console.log(`Found tool in cache: ${toolName} ${versionSpec} ${arch}`);
```

Comment:
console.log here seemed appropriate, let me know if I'm overlooking any reason we wouldn't want to do this here. I'm thinking we can add a silent option later if required.